src/core/mempool.rs

```diff
@@ -1166,7 +1212,35 @@ impl MemPoolDB {
 
         let tx_consideration_sampler = Uniform::new(0, 100);
         let mut rng = rand::thread_rng();
-        let mut remember_start_with_estimate = None;
+        let mut candidate_cache = CandidateCache::new();
+        let mut nonce_cache = NonceCache::new();
+
+        let sql = "
+             SELECT txid, origin_nonce, origin_address, sponsor_nonce, sponsor_address, fee_rate
+             FROM mempool
+             WHERE fee_rate IS NULL
+             ";
+        let mut query_stmt = self
+            .db
+            .prepare(&sql)
+            .map_err(|err| Error::SqliteError(err))?;
+        let mut null_iterator = query_stmt
+            .query(NO_PARAMS)
+            .map_err(|err| Error::SqliteError(err))?;
+
+        let sql = "
+            SELECT txid, origin_nonce, origin_address, sponsor_nonce, sponsor_address, fee_rate
+            FROM mempool
+            WHERE fee_rate IS NOT NULL
+            ORDER BY fee_rate DESC
```

Comment:
I was playing around with the mempool recently, and I noticed that adding `ORDER BY fee_rate` would slow down my queries considerably.  Can you check that we aren't doing a table scan with either of these queries via `EXPLAIN QUERY PLAN` on an instantiated database?  Thanks!

Reply:
With fee rate:
```
QUERY PLAN
`--SEARCH mempool USING INDEX by_fee_rate (fee_rate>?)
```
Without fee rate:
```
QUERY PLAN
`--SEARCH mempool USING INDEX by_fee_rate (fee_rate=?)
```

Reply:
nice

Reply:
thanks!